### PR TITLE
lib/model: Refactor file deletions when pulling

### PR DESCRIPTION
**Pull request by imsodin**
Tiny improvement split from #5680:  
Immediately delete symlinks and files, that are not candidates for renames. And reuse the file info from database if possible.

### CODE REVIEW

File: lib/model/folder_sendrecv.go [hunk 1]
```diff
@@ -333,17 +333,23 @@ func (f *sendReceiveFolder) processNeeded(dbUpdateChan chan<- dbUpdateJob, copyC
...
 				// WithNeed, furthermore, the file can simply be of the wrong
 				// type if we haven't yet managed to pull it.
 				if ok && !df.IsDeleted() && !df.IsSymlink() && !df.IsDirectory() && !df.IsInvalid() {
+					fileDeletions[file.Name] = file
 					// Put files into buckets per first hash
 					key := string(df.Blocks[0].Hash)
 					buckets[key] = append(buckets[key], df)
+				} else {
+					f.resetPullError(file.Name)
+					f.deleteFileWithCurrent(file, df, ok, dbUpdateChan, scanChan)
```
Comment (AudriusButkevicius) [inline lib/model/folder_sendrecv.go]: Almost feels like the delete should reset the error internally.

Comment (AudriusButkevicius) [reply lib/model/folder_sendrecv.go]: Also, this channel passing fiesta is getting out of hand, this whole thing is begging for a rewrite.

Comment (imsodin) [reply lib/model/folder_sendrecv.go]: To be consistent then all other ones should also go into e.g. `shortcutFile`, `handleDir` and so on. I opted for having all the error resets in the "top-level" functions (`processNeeded`, `processDeletions`). Honestly, I don't think it matters much.

And I most certainly don't care about channel passing here (and not much in general - or maybe, maybe not...).

Comment (imsodin) [reply lib/model/folder_sendrecv.go]: done

**Pull request merged**